### PR TITLE
 TST: switch to ubuntu yakkety for i386 testing

### PR DESCRIPTION
**Pull request by juliantaylor**

### CODE REVIEW

**Comment by juliantaylor** [comment]
urg, that should have gone to my fork for silent testing, please ignore for now

**Comment by juliantaylor** [comment]
hm a couple real failures
anyone know what this pip install warning means:

    warning: manifest_maker: standard file '-c' not found

**Comment by juliantaylor** [comment]
ok seems to be working now, so can be reviewed/merged
updates i386 test distribution to 16.10, newer python2, gcc, cython etc.
Also fixed the broken compiler warning check with a `-v` flag to pip install

File: tools/travis-test.sh [hunk 1]
```diff
@@ -8,7 +8,10 @@ export NPY_NUM_BUILD_JOBS=2
 # setup env
 if [ -r /usr/lib/libeatmydata/libeatmydata.so ]; then
   # much faster package installation
-  export LD_PRELOAD=/usr/lib/libeatmydata/libeatmydata.so
+  export LD_PRELOAD='/usr/lib/libeatmydata/libeatmydata.so'
+elif [ -r /usr/lib/*/libeatmydata.so ]; then
+  # much faster package installation
+  export LD_PRELOAD='/usr/$LIB/libeatmydata.so'
```
Comment (njsmith) [inline tools/travis-test.sh]: I don't understand what's going on here with `/usr/lib/*` versus `/usr/$LIB`?

File: tools/travis-test.sh [hunk 2]
```diff
@@ -102,7 +105,7 @@ setup_chroot()
     python-dev python-nose python-pip cython"
 
   # faster operation with preloaded eatmydata
-  echo /usr/lib/libeatmydata/libeatmydata.so | \
+  echo '/usr/$LIB/libeatmydata.so' | \
```
Comment (njsmith) [inline tools/travis-test.sh]: and it's odd that up there we had an `if` that decided what path to use for `libeatmydata.so`, but down here we always use `/usr/$LIB/libeatmydata.so`?

**Comment by njsmith** [comment]
Two comments in case they're useful. I'm not too bothered though, these travis scripts are always a bit arcane and fragile, and it doesn't matter too much as long as they work...

File: tools/travis-test.sh (hunk 1 above)
Comment (juliantaylor) [inline tools/travis-test.sh]: preload understands the special LIB variable, but we have to only enable it if it is available as our build will fail if gcc emits stderr (a bug we should fix) so I use a wildcard in the check, a little clunky but works

File: tools/travis-test.sh (hunk 2 above)
Comment (juliantaylor) [inline tools/travis-test.sh]: this is for inside the chroot, which is newer than trusty so always will use the multiarch path, when travis updates to xenial boxes the upper setting can be simplified too.
this setting should actually be done a bit earlier so we guarantee it is used for the dependency setup, done so now

**Comment by juliantaylor** [comment]
merging, as the earlier we get fixed tests the less PRs we get based on the not fixed tests..

**Pull request merged**